### PR TITLE
DEPR: ExtensionOpsMixin -> OpsMixin

### LINKED ISSUE #37080: DEPR: ExtensionOpsMixin in favor of OpsMixin
All our internal uses have been moved over except for the DecimalArray tests, which use ExtensionScalarOpsMixin.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #37080
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
thanks @jbrockmendel

**Comment by jorisvandenbossche** [comment]
Sorry, but this was not really ready:

- There has been no discussion about what the actual use case of `ExtensionScalarOpsMixin` (the issue mentions we don't use it internally, but it's goal is to serve external users, not internal)
- There is no indication in this PR of how to replace `ExtensionScalarOpsMixin` as a user of it
- The warning message suggests `OpsMixin` (but which is not actually a replacement), but this is not publicly exposed

I don't know if we should keep the class, but we should at least discuss it (and check if there is usage of it), and if we deprecate it give a clear reasoning and alternative.

**Comment by jreback** [comment]
@jorisvandenbossche ahh re-read your comment on the issue. if you want to revert ok, but to be honest this looks better / simpler.

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -487,6 +487,7 @@ Deprecations
 - Deprecated :meth:`Index.asi8` for :class:`Index` subclasses other than :class:`.DatetimeIndex`, :class:`.TimedeltaIndex`, and :class:`PeriodIndex` (:issue:`37877`)
 - The ``inplace`` parameter of :meth:`Categorical.remove_unused_categories` is deprecated and will be removed in a future version (:issue:`37643`)
 - The ``null_counts`` parameter of :meth:`DataFrame.info` is deprecated and replaced by ``show_counts``. It will be removed in a future version (:issue:`37999`)
+- :class:`ExtensionOpsMixin` and :class:`ExtensionScalarOpsMixin` are deprecated and will be removed in a future version.  Use ``pd.core.arraylike.OpsMixin`` instead (:issue:`37080`)
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.2.0.rst:490]: We shouldn't ever document something in `pandas.core` as  public. If this is going to change, it should be exposed  outside of `core`.

**Comment by TomAugspurger** [comment]
cc @Dr-Irv, who I believe implemented the mixin. I suspect you're using it?

**Comment by jorisvandenbossche** [comment]
I would say, let's revert it for now, until we have better discussed this. @jbrockmendel can you do that?

**Comment by jorisvandenbossche** [comment]
Actually, clicking the button is not that hard ;) -> https://github.com/pandas-dev/pandas/pull/38158

**Comment by jorisvandenbossche** [comment]
I reopened the issue, let's continue the discussion there: https://github.com/pandas-dev/pandas/issues/37080

**Comment by jreback** [comment]
> Actually, clicking the button is not that hard ;) -> https://github.com/pandas-dev/pandas/pull/38158

oh this must be new

**Pull request merged**